FirebaseStorage/Sources/Internal/StorageDeleteTask.swift

```diff
@@ -44,34 +44,34 @@ internal class StorageDeleteTask: StorageTask, StorageTaskManagement {
    * Prepares a task and begins execution.
    */
   internal func enqueue() {
-    weak var weakSelf = self
-    dispatchQueue.async {
-      guard let strongSelf = weakSelf else { return }
-      strongSelf.state = .queueing
-      var request = strongSelf.baseRequest
+    let completion = taskCompletion
+    taskCompletion = { (error: Error?) in
+      completion?(error)
+      // Reference self in completion handler in order to retain self until completion is called.
+      self.taskCompletion = nil
+    }
```

Comment:
Comment (no action required): Although this workaround for retaining the completion handler is arguably less readable at first glance, I couldn't come up with a cleaner alternative and think we should merge with this approach.

Your inline comments "`Reference self in completion handler in order to retain self until completion is called`", and using the same pattern throughout, address my readability concerns.